### PR TITLE
fix(node:https): mock `Agent` instead of throwing not implemented

### PR DESCRIPTION
**Pull request by vicb**
@pi0 This is my attempt to fix https://github.com/unjs/unenv/issues/295

I might have missed a few things, please let me know.

Thanks

### CODE REVIEW

File: src/runtime/node/https/index.ts [hunk 1]
```diff
@@ -1,26 +1,23 @@
 // https://nodejs.org/api/https.html
-import type nodeHttps from "node:https";
+import type https from "node:https";
```
[the remark this answers is not in the document]
Comment (pi0) [reply src/runtime/node/https/index.ts:2]:
```suggestion
import type nodeHttps from "node:https";
```

File: src/runtime/node/https/index.ts [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 import { notImplemented, notImplementedClass } from "../../_internal/utils";
 import mock from "../../mock/proxy";
 
-export const Server: typeof nodeHttps.Server =
+export const Server: typeof https.Server =
```
Comment (pi0) [inline src/runtime/node/https/index.ts:6]:
```suggestion
export const Server: typeof nodeHttps.Server =
```

File: src/runtime/node/https/index.ts [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
   notImplementedClass("https.Server");
 
-export const Agent: typeof nodeHttps.Agent = notImplementedClass("https.Agent");
+  export const Agent: typeof https.Agent = mock.__createMock__("https.Agent");
```
Comment (pi0) [inline src/runtime/node/https/index.ts:9]:
```suggestion
  export const Agent: typeof nodeHttps.Agent = mock.__createMock__("https.Agent");
```

File: src/runtime/node/https/index.ts [hunk 4]
```diff
@@ -1,26 +1,23 @@
...
+  export const globalAgent: typeof https.globalAgent = new Agent();
```
Comment (pi0) [inline src/runtime/node/https/index.ts:10]:
```suggestion
  export const globalAgent: typeof nodeHttps.globalAgent = new Agent();
```

File: src/runtime/node/https/index.ts [hunk 5]
```diff
@@ -1,26 +1,23 @@
...
-export const globalAgent =
-  (undefined as any as typeof nodeHttps.globalAgent) ||
-  mock.__createMock__("https.globalAgent");
-
-export const get = notImplemented<typeof nodeHttps.get>("https.get");
+export const get = notImplemented<typeof https.get>("https.get");
 
 export const createServer =
-  notImplemented<typeof nodeHttps.createServer>("https.createServer");
+  notImplemented<typeof https.createServer>("https.createServer");
```
Comment (pi0) [inline src/runtime/node/https/index.ts:15]:
```suggestion
  notImplemented<typeof nodeHttps.createServer>("https.createServer");
```

File: src/runtime/node/https/index.ts [hunk 6]
```diff
@@ -1,26 +1,23 @@
...
 
 export const request =
-  notImplemented<typeof nodeHttps.request>("https.request");
+  notImplemented<typeof https.request>("https.request");
```
Comment (pi0) [inline src/runtime/node/https/index.ts:18]:
```suggestion
  notImplemented<typeof nodeHttps.request>("https.request");
```

File: src/runtime/node/https/index.ts [hunk 7]
```diff
@@ -1,26 +1,23 @@
...
 
-export default <typeof nodeHttps>{
+export default <typeof https>{
```
Comment (pi0) [inline src/runtime/node/https/index.ts:20]:
```suggestion
export default <typeof nodeHttps>{
```

File: src/runtime/node/https/index.ts (hunk 2 above)
Comment (vicb) [reply src/runtime/node/https/index.ts]: Ok, I wanted to make it consistent with http.
Now I know the current way ;)
Thanks.

File: src/runtime/node/https/index.ts [hunk 8]
```diff
@@ -6,13 +6,10 @@ import mock from "../../mock/proxy";
...
-export const Agent: typeof nodeHttps.Agent = notImplementedClass("https.Agent");
+export const Agent: typeof nodeHttps.Agent = mock.__createMock__("https.Agent");
+export const globalAgent: typeof nodeHttps.globalAgent = new Agent();
 
-export const globalAgent =
-  (undefined as any as typeof nodeHttps.globalAgent) ||
-  mock.__createMock__("https.globalAgent");
-
-export const get = notImplemented<typeof nodeHttps.get>("https.get");
+export const get = notImplemented<typeof https.get>("https.get");
```
Comment (vicb) [inline src/runtime/node/https/index.ts:12]:
```suggestion
-export const get = notImplemented<typeof https.get>("https.get");
+-export const get = notImplemented<typeof nodeHttps.get>("https.get");
```

**Comment by pi0** [comment]
@vicb i will manually resolve conflicts locally on it now.

**Pull request merged**